File: stringCompressionIII.py
class Solution:
    def __init__(self) -> None:
        pass

    def compressedString(self, word: str) -> str:
        words = list(word)
        ans = ""
        i = 0
        while i < len(words):
            count = 0
            curr = words[i]
            while i < len(words) and words[i] == curr and count < 9:
                i+=1
                count+=1
            ans += str(count) + curr
        
        return ans
    
def main():
    solution = Solution()
    print(solution.compressedString("aaaaaaaaaaaaaabb"))

if __name__ == "__main__":
    main()

File: StringCompressionIII.java
public class StringCompressionIII {
    public String compressedString(String word) {
        StringBuilder sb = new StringBuilder();
        int i = 0;
        char curr = word.charAt(0);
        while (i < word.length()) {
            int count = 1;
            curr = word.charAt(i);
            while(++i < word.length() && word.charAt(i) == word.charAt(i-1) && count < 9) {
                count++;
            }
            sb.append(count);
            sb.append(curr);
        }

        return sb.toString();
    }
    public static void main(String[] args) {
        String word = "aaaaaaaaaaaaaabb";
        StringCompressionIII stringCompressionIII = new StringCompressionIII();
        System.out.println(stringCompressionIII.compressedString(word));
    }
}
